metrics/ci_integrations/docs/07_sync_performance_investigation.md

```diff
@@ -0,0 +1,161 @@
+
+So, we admit that the one-build synchronization executes in a reasonable time. However, the algorithm requires review for different integrations on syncing a big number of builds.
+
+## Sync Command Performance in CI
+
+The above section examines the sync execution time for one-build synchronization and admits this time as reasonable. However, the performed experiments demonstrate results for running the CI Integration tool on a local machine. In reality, if we want to [automate builds synching](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/02_ci_integration_user_guide.md#automating-ci-integrations), we face the sync-time problem. In the below subsections, we discover the problem for each CI integration and try to figure out what causes the problem.
+
+### GitHub Actions
+
+The GitHub Actions job that synchronizes builds performs the following steps:
+
+1. Set up job.
+2. Checkout the repository.
+3. Download the CI Integration tool.
+4. Wait for the build to finish.
+5. Apply environment variables to the configuration file.
+6. Synchronize builds data.
+7. Post Checkout clean up.
+8. Complete job.
+
+The initial step and the two last are performed automatically - the job specification lists only steps from 2 to 6. Also, we should note that the 4th step is performed only if the sync workflow should sync the finished build. Otherwise, the 3rd step is skipped. More precisely, this means that if the workflow is dispatched with the `build_started` event, the job shouldn't wait for a build to finish as the sync is performed for the running build. On the other hand, if the workflow is dispatched with the `build_finished` event, the job should wait for the build to finish before performing sync.
+
+Although the job performs two additional steps after the sync is finished, they don't matter as the build is already synced. The steps that precede the synchronization influence the time spent for a new build to be available in the destination database and appear on the Metrics Web Application. Therefore, we focus on steps from the initial to the sync step itself (1-6).
+
+The following table contains the information about the execution time of each step of the sync job. The original sample contains eleven objects representing a single job run each. Please note that the value `0 s` (zero seconds) means `close to 0s` - the reason is that GitHub Actions API and web UI don't provide the precise value in milliseconds.
+
+||Setup|Checkout|Download|Wait|Configure|Sync|
+|---|---|---|---|---|---|---|
+|_Average_|7,18 s|12,91 s|1,91 s|1,27 s|0,18 s|3,27 s|
+|_Trimmed mean_|5,44 s|9,11 s|1,89 s|0,44 s|0,11 s|3,33 s|
+|_Max_|28 s|58 s|4 s|10 s|1 s|4 s|
+|_Min_|2 s|2 s|0 s|0 s|0 s|2 s|
+|_Median_|3 s|3 s|2 s|0 s|0 s|3 s|
+|_Mode_|3 s|3 s|1 s|0 s|0 s|4 s|
```

Comment:
Would it be great to have a `Total` (or something like this) column in this table representing the total duration taken to run the `sync job`?